resolwe/flow/managers/listener.py

```diff
@@ -501,7 +501,14 @@ def handle_download_finished(self, obj):
                 obj, {ExecutorProtocol.RESULT: ExecutorProtocol.RESULT_ERROR},
             )
         else:
-            query.update(status=StorageLocation.STATUS_DONE)
+            # Add files to the downloaded storage location.
+            storage_location = query.get()
+            default_storage_location = (
+                storage_location.file_storage.default_storage_location
```

Comment:
The code (indentation) would be nicer if you rename `storage_location` to `location`, but also probably less consistent and self-explanatory.